### DISCUSSION TITLE
Upsert using multikey unique constraint

### DISCUSSION THREAD

**Discussion by aakashkag** [prisma/prisma#2557] [Q&A] [answered]
OS: Ubuntu 19.10
Database: Postgres,
Prisma version: @prisma/cli: 2.0.0-beta.5
Node.js version: v12.14.0

Hi
I am trying to upsert data.
I added unique constraints on combination of (user_id + organization_id)

```
model User {
  user_id                            Int              @default(autoincrement()) @id
  organization_id                    Int
  user_email                         String
@@unique([user_email, organization_id], name: "User_user_email_organization_id_key")
}
```

 ```
ctx.prisma.user.upsert({
            where: {
                organization_id: organization_id,
                user_email: args.user_email,
            },
            create: args,
            update: {},
        });

```

I am getting following error
 "Unknown arg `organization_id` in where.organization_id for type UserWhereUniqueInput. Did you mean `user_id`? Available args:",

 "Unknown arg `user_email` in where.user_email for type UserWhereUniqueInput. Did you mean `user_id`? Available args:",

If i just keep unique constrain on only one field and apply only one condition in where clause then it's working fine.

**Comment by pantharshit00** [comment, marked as answer]
You will need to perform the following query, you are missing a sub key:
```ts
const data = await client.user.upsert({
    where: {
      User_user_email_organization_id_key: {
        user_email: "",
        organization_id: 0,
      },
    },
    create: args,
    update: {},
  });
```

**Comment by ludralph** [comment]
Hi @aakashkag,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
